### PR TITLE
Beta branch and Semantic Release

### PR DESCRIPTION
**Pull request by dbouwman**
1. Description:

This adds the `beta` branch, and automated release tooling based on `multi-semantic-release`.

Basically, merges to `master` will cut a formal release, merges to `beta` will cut a `v<next>-beta.N` release

The main difference in the repo will now have tags for each package:
image

Additionally, each package will have it's own CHANGELOG that's automatically managed.

1. Instructions for testing:

Since this cuts actual releases, we want to do the testing against the `beta` branch as that will cut `beta-N` releases.

Basically pull beta, make some changes in a package, use semantic commit messages (should be enforced) and push up to beta. You'll see the `release` workflow run, and if you made actual changes, it will cut a release.

### Version Numbers
By default, the tooling will manage version numbers independently, but I believe I have configured things so a change to any package will release all packages at that same version number. 🤞🏻

1. [x] ran commit script (`npm run c`)

### CODE REVIEW

**Comment by dbouwman** [comment]
One thing we should consider is loading in all the old CHANGELOG entries into the package specific changelogs, and changing the root CHANGELOG to just have links to the package changelogs which is what I did in templates.js -> https://github.com/Esri/templates-js/blob/beta/CHANGELOG.md

File: packages/common/test/search/utils.test.ts [hunk 1]
```diff
@@ -205,7 +205,13 @@ describe("Search Utils:", () => {
         const chk = relativeDateToDateRange(rd);
         expect(chk.from).toBeLessThan(chk.to);
         const m = new Date(chk.from).getMonth();
-        expect(m).toBeLessThan(new Date().getMonth());
+        const curMonth = new Date().getMonth();
+        // Account for January/December
+        if (curMonth > 0) {
+          expect(m).toBeLessThan(new Date().getMonth());
```
Comment (rweber-esri) [inline packages/common/test/search/utils.test.ts:211]: Minor, could just use `curMonth`

```js
expect(m).toBeLessThan(curMonth);
```

**Review by rweber-esri — APPROVED** [review]
I did not test this, but looks looks good to me. Just a couple minor comments/questions.

File: package.json [hunk 2]
```diff
@@ -74,70 +90,71 @@
...
 		"build:tests": "npm run build:esm && npm run build:node",
-		"build:esm:es5": "lerna run build:esm:es5",
-		"build:esm": "lerna run build:esm",
-		"build:umd": "lerna run build:umd",
-		"build:node": "lerna run build:node",
-		"clean": "npm run clean:dist & npm run clean:deps",
-		"clean:dist": "rm -rf packages/*/dist/",
+		"build:umd": "npx ultra --build -r filter=\"packages/*\"  build:umd",
+		"build": "npm run build:esm && npm run build:esm:es5 && npm run build:umd && npm run build:node",
+		"c": "npm run precommit && git-cz",
```
[the remark this answers is not in the document]
Comment (dbouwman) [reply package.json:108]: Added that to the README.md - at some point I'll add it into the doc site

File: packages/common/test/search/utils.test.ts (hunk 1 above)
Comment (dbouwman) [reply packages/common/test/search/utils.test.ts:211]: eagle-eyes++ :)

**Review by mjuniper — APPROVED** [review]
LGTM, thanks for doing this!

**Review by tomwayson — APPROVED** [review]
this is going to be awesome, thanks @dbouwman!
